src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ArrayOfEmbeddedDataNode.cs

```diff
@@ -23,12 +23,14 @@ public class ArrayOfEmbeddedDataNode<TEmbedded> : ObjectNode
         private ObjectAndOffsetSymbolNode _startSymbol;
         private ObjectAndOffsetSymbolNode _endSymbol;
         private IComparer<TEmbedded> _sorter;
+        private EmbeddedObjectNode _endOfArraySentinel;
```

Comment:
Is this really general purpose enough to have in here? Can you give me other examples where this will be used?